src/main/java/com/googlecode/protobuf/format/JsonFormat.java

```diff
@@ -875,6 +885,36 @@ protected void mergeField(Tokenizer tokenizer,
             mergeField(tokenizer, extensionRegistry, builder);
         }
     }
+    
+    private void handleUnknownField(Tokenizer tokenizer,    										
+    		Message.Builder builder,
+    		int fieldNumber) throws ParseException {    	
+    	tokenizer.tryConsume(":");
+    	if ("{".equals(tokenizer.currentToken())) {
+    		// Message structure
+    		tokenizer.consume("{");
+    		do {
+    			tokenizer.consumeIdentifier();
+    			handleUnknownField(tokenizer, builder, fieldNumber);
+    		} while (tokenizer.tryConsume(","));
+    		tokenizer.consume("}");
+    	} else if ("[".equals(tokenizer.currentToken())) {
+    		// Collection
+    		tokenizer.consume("[");
+    		do {
+    			handleUnknownField(tokenizer, builder, fieldNumber);
+    		} while (tokenizer.tryConsume(","));
+    		tokenizer.consume("]");
+    	} else {
+    		ByteString data = tokenizer.consumeByteString();
+    		UnknownFieldSet.Builder unknownBuilder = builder.getUnknownFields().toBuilder();
+    		unknownBuilder.addField(fieldNumber, UnknownFieldSet.Field
+    				.newBuilder()
+    				.addLengthDelimited(data)

```

Comment:
How is the data type known to be lengthDelimited? Could it be varint or fixed32/64?